.github/workflows/.reusable-integration-test.yml

```diff
@@ -164,6 +164,81 @@ jobs:
         run: |
           kubectl logs -n connaisseur -lapp.kubernetes.io/name=connaisseur --prefix=true --tail=-1
 
+  self-hosted-notary:
+    name: self-hosted-notary
+    runs-on: ubuntu-latest
+    if: |
+      inputs.skip_integration_tests != 'non-required' &&
+      inputs.skip_integration_tests != 'all'
+    permissions:
+      packages: read
+    env:
+      IMAGE: ${{ inputs.build_image_repository }}
+      TAG: ${{ inputs.build_tag }}
+      COSIGN_PUBLIC_KEY: ${{ inputs.cosign_public_key }}
+    strategy:
+      fail-fast: false
+      matrix:
+       integration-test-arg:
+        [
+          "self-hosted-notary"
+        ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Login with registry
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+        with:
+          registry: ${{ inputs.build_registry }}
+          username: ${{ inputs.repo_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install yq
+        run: |
+          sudo snap install yq
+      - name: Setup notary signer instance
+        run: |
+          docker run -d -p 7899:7899 -v ./tests/data/notary_service_container/signer:/etc/docker/notary-signer/ notary:signer -config=/etc/docker/notary-signer/config.json
+      - name: Get notary signer instance IP
+        id: get_notary_signer_ip
+        uses: ./.github/actions/notary-signer-ip
+      - name: Setup notary server instance
+        run: |
+          echo ${{ steps.get_notary_signer_ip.outputs.notary_signer_ip }}
+          docker run -d -p 4443:4443 --add-host notary.signer:${{ steps.get_notary_signer_ip.outputs.notary_signer_ip }} -v ./tests/data/notary_service_container/server:/etc/docker/notary-server notary:server -config=/etc/docker/notary-server/config.json -logf=json
+      - name: Get container IPs
+        id: get_notary_server_ip
+        uses: ./.github/actions/notary-server-ip
+      - name: Populate notary instance with trust data
+        uses: ./.github/actions/setup-notary
+        id: setup_notary
+        env:
+          NOTARY_IP: ${{ steps.get_notary_server_ip.outputs.notary_ip }}
+      - uses: ./.github/actions/k8s-version-config
+        name: Setup k8s cluster
+        with:
+          k8s-version: v1.25
```

Comment:
Arbitrary choice or is that the oldest supported version?

Reply:
neither nor I'd say :grin: just copy paste from the general `integration-test` job. BUT: Really good point, bc 1.25 is EOL since end of october, so I think we should change, but I don't know which version is best for testing. Opinion?

https://kubernetes.io/releases/patch-releases/#non-active-branch-history